packages/fx-core/src/plugins/solution/fx-solution/solution.ts

```diff
@@ -1512,21 +1543,24 @@ export class TeamsAppSolution implements Solution {
           : "undefined",
       });
 
-      return ok(permissions);
+      return ok({
+        state: CollaborationState.OK,
+        permissions,
+      });
     } finally {
       ctx.envInfo.profile.get(GLOBAL_CONFIG)?.delete(USER_INFO);
```

Comment:
Seems like you are using the config interface as an message passing mechanism... which is not config interface was intended to be used.

Reply:
Thanks, as discussed offline, we can add parameters to plugin to not use config. I will add a task for this